frontend/packages/console-shared/src/hooks/useLastNamespace.ts

```diff
@@ -0,0 +1,7 @@
+import { useContext } from 'react';
+import { NamespaceContext } from '@console/app/src/components/detect-namespace/namespace';
+
+export const useLastNamespace = (): [string, (ns: string) => void] => {
```

Comment:
Probably should call this `useActiveNamespace` since that's the current term devs will be familiar with.